PlagiarismPlugin.inc.php

```diff
@@ -43,172 +75,431 @@ public function getDescription() {
 	/**
 	 * @copydoc LazyLoadPlugin::getCanEnable()
 	 */
-	function getCanEnable($contextId = null) {
+	public function getCanEnable($contextId = null) {
 		return !Config::getVar('ithenticate', 'ithenticate');
 	}
 
 	/**
 	 * @copydoc LazyLoadPlugin::getCanDisable()
 	 */
-	function getCanDisable($contextId = null) {
+	public function getCanDisable($contextId = null) {
 		return !Config::getVar('ithenticate', 'ithenticate');
 	}
 
 	/**
 	 * @copydoc LazyLoadPlugin::getEnabled()
 	 */
-	function getEnabled($contextId = null) {
+	public function getEnabled($contextId = null) {
 		return parent::getEnabled($contextId) || Config::getVar('ithenticate', 'ithenticate');
 	}
 
+	/**
+	 * Add the IThenticate EULA url as a checklist of submission process
+	 * 
+	 * @param string $hookName `TemplateManager::display` or `TemplateManager::fetch`
+	 * @param array $params
+	 * 
+	 * @return bool
+	 */
+	public function addEulaToChecklist($hookName, $params) {
+		$templateManager = & $params[0]; /** @var TemplateManager $templateManager */
+		$context = $templateManager->getTemplateVars('currentContext'); /** @var Context $context */
+		
+		if (!$context || strtolower($templateManager->getTemplateVars('requestedPage') ?? '') !== 'submission') {
+			return false;
+		}
+
+		$eualDetails = $this->getContextEulaDetails($context);
+		
+		foreach($context->getData('submissionChecklist') as $locale => $checklist) {
+			array_push($checklist, [
+				'order' => (collect($checklist)->pluck('order')->sort(SORT_REGULAR)->last() ?? 0) + 1,
+				'content' => __('plugins.generic.plagiarism.submission.checklist.eula', [
+					'localizedEulaUrl' => $eualDetails[$locale]['url']
+				]),
+			]);
+
+			$context->setData('submissionChecklist', $checklist, $locale);
+		}
+
+		$templateManager->assign('currentContext', $context);
```

Comment:
Is this necessary? I'm pretty sure the code will operate on the existing `Context` object rather than on a copy of it.

Reply:
right, not needed . removed